core_extract_comments.py

```diff
@@ -40,9 +42,16 @@ def get_comments_with_product_id(product_id):
         return reviews
     if not re.match('^[A-Z0-9]{10}$', product_id):
         return reviews
-    for page_number in range(100):
-        product_reviews_link = get_product_reviews_url(product_id, page_number)
-        so = get_soup(product_reviews_link)
+
+    product_reviews_link = get_product_reviews_url(product_id)
+    so = get_soup(product_reviews_link)
+    max_page_number = so.find_all("li", {'class': 'page-button'})
+    max_page_number = int(max_page_number[-1].text) if max_page_number else 1
+
+    for page_number in range(1, max_page_number + 1):
+        if page_number > 1:
```

Comment:
What is the reason for this condition here: page_number > 1?

It's equivalent to range(2, ...) no?

Reply:
@NickyFot

Reply:
yes! basically, because you set so earlier on (line 47) for the first page there is no point making another call in the loop (reducing the # calls to amazon and therefore the chance to get detected!)

Reply:
@NickyFot got it! Thanks!

Reply:
@philipperemy the if only refers to the call, scrapping comments for page 1 still happens within the loop

Reply:
Looks great thanks!